contracts/distribution/subdomains/pricing/ZNSAsymptoticPricing.sol

```diff
@@ -0,0 +1,208 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.18;
+
+import { AZNSPricingWithFee } from "../abstractions/AZNSPricingWithFee.sol";
+import { StringUtils } from "../../../utils/StringUtils.sol";
+import { IZNSRegistry } from "../../../registry/IZNSRegistry.sol";
+// TODO sub: possibly remove if moved to an interface
+import { IDomainPriceConfig } from "../../../abstractions/IDomainPriceConfig.sol";
+// TODO sub: do we need this ??
+import { AAccessControlled } from "../../../access/AAccessControlled.sol";
+import { ARegistryWired } from "../../../abstractions/ARegistryWired.sol";
+
+
+// TODO sub: figure out how to interface here with the abstract and PriceConfig struct !!
+contract ZNSAsymptoticPricing is AAccessControlled, ARegistryWired, AZNSPricingWithFee, IDomainPriceConfig {
+    using StringUtils for string;
+
+    // TODO sub: possibly move to an interface
+    event MaxPriceSet(bytes32 indexed domainHash, uint256 price);
+    event MinPriceSet(bytes32 indexed domainHash, uint256 price);
+    event BaseLengthSet(bytes32 indexed domainHash, uint256 length);
+    event MaxLengthSet(bytes32 indexed domainHash, uint256 length);
+    event PrecisionMultiplierSet(bytes32 indexed domainHash, uint256 precision);
+    event FeePercentageSet(bytes32 indexed domainHash, uint256 feePercentage);
+    event PriceConfigSet(
+        bytes32 indexed domainHash,
+        uint256 maxPrice,
+        uint256 minPrice,
+        uint256 maxLength,
+        uint256 baseLength,
+        uint256 precisionMultiplier,
+        uint256 feePercentage
+    );
+
+    uint256 public constant PERCENTAGE_BASIS = 10000;
+
+    mapping(bytes32 => DomainPriceConfig) public priceConfigs;
+
+
+    constructor(address _accessController, address _registry) {
+        _setAccessController(_accessController);
+        setRegistry(_registry);
+    }
+
+    function getPrice(
+        bytes32 parentHash,
+        string calldata label
+    ) public view override returns (uint256) {
+        uint256 length = label.strlen();
```

Comment:
Don't set this local variable just use label.strlen() on 51

Reply:
good catch, but I believe, the wrong one. later in the code we call the strlen() again, which is an error. we can just use this var, since it's used twice. that's better than calling the lib twice, I believe.

Reply:
yes, its not being used on 53
you should just delete 49 and use label.strlen() on 51 too though

worth a sanity test at least.

Reply:
you really hate local vars, huh? :)) even two identical calls is better? :)

Reply:
It depends on what the call is so usually just test it if it's in question.
I understand you did some testing of it so all good.